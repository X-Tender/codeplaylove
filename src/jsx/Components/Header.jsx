import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
	subhead: PropTypes.string.isRequired,
	headline: PropTypes.string.isRequired,
	className: PropTypes.string,
};

const defaultProps = {
	className: '',
};

const Header = ({ subhead, headline, className }) => {
	const classes = `header ${className}`;
	return (
		<header className={classes}>
			<h3 className="introduction__subhead">{subhead}</h3>
			<h2 className="introduction__headline">{headline}</h2>
		</header>
	);
};

Header.propTypes = propTypes;
Header.defaultProps = defaultProps;

export default Header;
