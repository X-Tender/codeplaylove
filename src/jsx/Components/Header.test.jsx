import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Header', () => {
	const subhead = 'My subhead';
	const headline = 'My headline';
	const className = 'my-class';

	describe('with class', () => {
		const headerWithClass = shallow(
			<Header className={className} headline={headline} subhead={subhead} />
		);

		it('renders correctly', () => {
			expect(headerWithClass).toMatchSnapshot();
		});

		it('renders the correct props', () => {
			expect(headerWithClass.find(`header.${className}`).exists()).toBe(true);
			expect(headerWithClass.find('.introduction__subhead').text()).toEqual(subhead);
			expect(headerWithClass.find('.introduction__headline').text()).toEqual(headline);
		});
	});

	describe('without class', () => {
		const headerWithoutClass = shallow(<Header headline={headline} subhead={subhead} />);

		it('renders correctly', () => {
			expect(headerWithoutClass).toMatchSnapshot();
		});

		it('renders the correct props', () => {
			expect(headerWithoutClass.find('header').prop('className')).toEqual('header ');
			expect(headerWithoutClass.find('.introduction__subhead').text()).toEqual(subhead);
			expect(headerWithoutClass.find('.introduction__headline').text()).toEqual(headline);
		});
	});
});
