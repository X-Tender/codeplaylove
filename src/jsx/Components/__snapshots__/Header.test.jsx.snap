// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header with class renders correctly 1`] = `
<header
  className="header my-class"
>
  <h3
    className="introduction__subhead"
  >
    My subhead
  </h3>
  <h2
    className="introduction__headline"
  >
    My headline
  </h2>
</header>
`;

exports[`Header without class renders correctly 1`] = `
<header
  className="header "
>
  <h3
    className="introduction__subhead"
  >
    My subhead
  </h3>
  <h2
    className="introduction__headline"
  >
    My headline
  </h2>
</header>
`;
